### DISCUSSION TITLE
Understanding FreeDesktop.org Secret Service Integration

### DISCUSSION THREAD

**Discussion by Offerel** [keepassxreboot/keepassxc#8016] [Q&A] [answered]
I have enabled FreeDesktop.org Secret Service integration in KeepassXC in XFCE/Manjaro,. It works basically but i do not understand completely how it works. Fox example, i have an existing entry for my E-Mail. But after enabling Secret Service integration instead of re-using the existing entry for E-Mail, Evolution creates a new entry with the password and some advanced fields.

Is it however possible to re-use the existing entry for E-Mail? Do i have to add the advanced fields from the new created entry to my E-Mail entry or is it identified by the title from the new entry? The same is asked for some GPG entries...

**Comment by Aetf** [comment, marked as answer]
The secret service integration will use a separate group that you configured to save entries, so by default non of the existing entries are exposed unless you choose to use an existing group.

Also, applications expect to find entries in application-specific formats (i.e. certain attributes for lookup, certain format in the password field, etc.) It's totally up to the application and varies across applications.

Therefore, it's generally advised to not create those entries manually. Instead, first run the application and let it create the entry first, then, you can edit the newly created entry and use reference fields to reference the password or other fields in your existing entry.

**Reply by Offerel** [reply]
Ahhh ok, now I understand. I think this explanation should be inserted into the documentation in some manor. Many thx for clarification.

**Reply by Aetf** [reply]
Yeah, there's virtually no documentation for this part yet. But I'll take note when I eventually get to document the secret service integration :)
